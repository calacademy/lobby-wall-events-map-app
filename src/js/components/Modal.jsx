'use strict'

var React = require('react')
var ReactCSSTransitionGroup = require('react-addons-css-transition-group')

var Modal = React.createClass({
    render: function() {
        if(this.props.isOpen){
            return (
              <ReactCSSTransitionGroup
                transitionName={this.props.transitionName}
                transitionEnterTimeout={500}
                transitionLeaveTimeout={500}>
                {this.props.children}  
              </ReactCSSTransitionGroup>
            );
        } else {
            return (
              <ReactCSSTransitionGroup
                transitionName={this.props.transitionName}
                transitionEnterTimeout={500}
                transitionLeaveTimeout={500}>
              </ReactCSSTransitionGroup>
            )
        }
    }
})

export default Modal
