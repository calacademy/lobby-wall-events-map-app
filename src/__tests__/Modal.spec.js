import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Modal from '../js/components/Modal'

describe('Modal', () => {

  let modal

  beforeEach(() => {
    modal = shallow(<Modal transitionName="foobar" />)
  })

  // regression test - jest snapshot
  it('enzyme shallow matches snapshot', () => {
    expect(shallowToJson(modal)).toMatchSnapshot()
  })

})
