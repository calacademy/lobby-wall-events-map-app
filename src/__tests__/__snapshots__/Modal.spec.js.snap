exports[`Modal enzyme shallow matches snapshot 1`] = `
<ReactCSSTransitionGroup
  transitionAppear={false}
  transitionEnter={true}
  transitionEnterTimeout={500}
  transitionLeave={true}
  transitionLeaveTimeout={500}
  transitionName="foobar" />
`;
